### PR TITLE
Resolving sibling modules with absolute imports

### PR DESCRIPTION
**Pull request by linzhp**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

`py_test` and `py_binary` unconditionally depends on `py_library` in the same package. In order to do that, the Gazelle plugin introduce an virtual import to identify the `py_library` target.

## What is the new behavior?
With the capability of resolving sibling modules introduced in #1027, we can resolve `py_library` as needed in `py_test` and `py_binary`, and we no longer need any virtual imports or id for `py_library` targets. This simplifies the logic.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: gazelle/python/testdata/simple_test_with_conftest/README.md [hunk 1]
```diff
@@ -1,4 +1,3 @@
 # Simple test with conftest.py
 
-This test case asserts that a simple `py_test` is generated as expected when a
-`conftest.py` is present.
+This test case asserts that a simple `py_test` does not depend on conftest when it doesn't import it.
```
Comment (f0rmiga) [inline gazelle/python/testdata/simple_test_with_conftest/README.md:3]: conftest files don't need to be imported. They should be added unconditionally.

**Review by f0rmiga — CHANGES_REQUESTED** [review]

**Comment by linzhp** [comment]
Ready for review, but the CI is affected by https://github.com/orgs/community/discussions/45830

File: gazelle/python/target.go [hunk 2]
```diff
@@ -15,12 +15,11 @@
 package python
 
 import (
-	"path/filepath"
-
 	"github.com/bazelbuild/bazel-gazelle/config"
 	"github.com/bazelbuild/bazel-gazelle/rule"
 	"github.com/emirpasic/gods/sets/treeset"
 	godsutils "github.com/emirpasic/gods/utils"
+	"path/filepath"
```
Comment (f0rmiga) [inline gazelle/python/target.go:22]: Unnecessary and wrong change, we keep standard lib imports at the top.

File: gazelle/python/testdata/generated_test_entrypoint/BUILD.out [hunk 3]
```diff
@@ -17,8 +17,5 @@ py_test(
     name = "generated_test_entrypoint_test",
     srcs = [":__test__"],
     main = ":__test__.py",
-    deps = [
-        ":__test__",
-        ":generated_test_entrypoint",
-    ],
+    deps = [":__test__"],
```
Comment (f0rmiga) [inline gazelle/python/testdata/generated_test_entrypoint/BUILD.out:20]: This may be a breaking change.

**Review by f0rmiga — APPROVED** [review]

File: gazelle/python/target.go (hunk 2 above)
Comment (linzhp) [reply gazelle/python/target.go:22]: That was changed by my GoLand and I didn't notice.

**Pull request merged**